### PR TITLE
Fix pylint errors and warnings

### PR DESCRIPTION
**Pull request by matthewrmshin**
Should eliminate all errors and most warnings.

### CODE REVIEW

File: isodatetime/data.py [hunk 1]
```diff
@@ -1703,12 +1700,11 @@ def get_is_leap_year(year):
 
 def get_days_in_year_range(start_year, end_year):
     """Return the number of days within this year range (inclusive)."""
-    return _get_days_in_year_range(start_year, end_year,
-                                   calendar_mode=CALENDAR.mode)
+    return _get_days_in_year_range(start_year, end_year, CALENDAR.mode)
```
Comment (oliver-sanders) [inline isodatetime/data.py]: Do we still need the `CALENDAR.mode` argument?

File: isodatetime/data.py [hunk 2]
```diff
@@ -1048,10 +1044,10 @@ def apply_time_zone_offset(self, offset):
                     offset.minutes / float(CALENDAR.MINUTES_IN_HOUR))
             else:
                 self.minute_of_hour += offset.minutes
-            self._tick_over()
+            self.tick_over()
```
Comment (oliver-sanders) [inline isodatetime/data.py]: Why have some methods changed from private to public?

File: isodatetime/parser_spec.py [hunk 3]
```diff
@@ -285,7 +285,13 @@ def __str__(self):
 
 
 def get_date_translate_info(num_expanded_year_digits=2):
-    expanded_year_digit_regex = "\d" * num_expanded_year_digits
+    """Return list of 4-element tuples with date translate information.
+
+    Each tuple is (
+        basic regex, regex with capture group, format substitution,
+        property name)
+    """
```
Comment (oliver-sanders) [inline isodatetime/parser_spec.py]: Would be good if we could stick to a standard docstring format e.g: http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

I.E. something like...

```
returns:
    list: List of four element tuples with the following components.
        - basic regex (str) - Some description text
        - regex with capture group (str) - ...
        ...
```

File: isodatetime/data.py [hunk 4]
```diff
@@ -1390,7 +1385,7 @@ def _add_months(self, num_months):
             if self.get_is_week_date():
                 was_week_date = True
             self.to_calendar_date()
-        for i in range(abs(num_months)):
+        for _ in range(abs(num_months)):
```
Comment (dvalters) [inline isodatetime/data.py]: I get why PyLint suggests this, but I find it less readable/intuitive. (Oh well)

File: isodatetime/data.py [hunk 5]
```diff
@@ -1750,11 +1746,11 @@ def _get_days_in_year_range(start_year, end_year, calendar_mode=None):
 
 def get_days_in_year(year):
     """Return the number of days in this particular year."""
-    return _get_days_in_year(year, calendar_mode=CALENDAR.mode)
+    return _get_days_in_year(year, CALENDAR.mode)
 
 
 @util.cache_results
-def _get_days_in_year(year, calendar_mode=None):
+def _get_days_in_year(year, _):
```
Comment (dvalters) [inline isodatetime/data.py]: If the argument isn't used, why does it need a 'discardable' argument passing? (I must be missing something here...)

Comment (matthewrmshin) [inline isodatetime/data.py]: It is used by `util.cache_results`.

File: isodatetime/data.py (hunk 1 above)
Comment (matthewrmshin) [inline isodatetime/data.py]: It is used by `util.cache_results`.

File: isodatetime/data.py (hunk 2 above)
Comment (matthewrmshin) [inline isodatetime/data.py]: In `TimePoint.add_truncated`, the method is invoked as `new.tick_over`, which is a bit bad according to pylint.

File: isodatetime/parser_spec.py (hunk 3 above)
Comment (matthewrmshin) [inline isodatetime/parser_spec.py]: Done.

**Pull request merged**